cloudfour/cloudfour.com-patterns
# Improve support in IE11 and other lacking browsers

## Overview

Closes #1096 

This improves the experience of using the cloud four site in IE11 and other lacking browsers.

@tylersticka a big thing that wasn't supported was logical properties.

Ideally code like this:
```css
.something {
  inline-size: 200px;
}
```

would be transpiled like this:

```css
.something {
  width: 200px;
}
@supports (inline-size: 200px) {
  inline-size: 200px;
  width: initial;
}
```

Unfortunately, the existing [postcss plugin](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-logical) instead transforms it like this:

```css
.something {
  width: 200px;
}
```

So it can sort of downgrade the code but that's all. Even though this is less-than-ideal, I figured that this is an ok solution because it drastically improves the experience for IE11 users, fixing a whole bunch of issues with very little effort.
If in the future we decide to have some sort of translations/other language content, we could either drop this plugin or send a PR to the plugin to automatically generate `@supports` blocks.

Testing:

- Get your local wp running
- Link your `@cloudfour/patterns` in the wp site to your local copy:
  - Go to your local repo copy of `@cloudfour/patterns`
  - `git checkout browser-support-improvements`
  - `npm run build`
  - `npm link` - this sets up a global symlink to this copy of the repo
  - Go to your local wp folder
  - `cd app/public/wp-content/themes/cloudfour2021`
  - `npm link @cloudfour/patterns` - this sets up that folder in node_modules to be symlinked to your local copy of the repo.
- Hard-refresh pages in the browser to make sure the new CSS is read through the symlinks
- Test in IE11 that at least the problems listed in the issue were fixed
- Use the "live link" feature in Local to test in Opera Mini with the data savings setting set to "extreme"
